src/main/groovy/wooga/gradle/paket/base/utils/internal/PaketUPMWrapperReference.groovy

```diff
@@ -0,0 +1,55 @@
+package wooga.gradle.paket.base.utils.internal
+
+import org.gradle.api.Project
+
+/*
+ * Copyright 2022 Wooga GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+class PaketUPMWrapperReference {
+    public String upmPackageNameUnversioned
+    public String upmPackageName
+    public String upmPackageURL;
+    public File upmPackageTar
+    public File file;
+    public boolean exists
+
+    public final static String upmWrapperReferenceFile = "paket.upm.wrapper.reference"
+
+    public static boolean IsReferenceFile(File file)
+    {
+        return file.name.equals(upmWrapperReferenceFile)
+    }
+
+    PaketUPMWrapperReference(File wrapperReferencesFile) {
+        file = wrapperReferencesFile
+        exists = false
+        if (file.exists()) {
+            exists = true
+            def parts = file.text.trim().split(";")
+            upmPackageURL = parts[0]
+            upmPackageName  = parts[1]
+            upmPackageNameUnversioned = upmPackageName.split("@")[0]
+            upmPackageTar = new File ("${file.parentFile.path}/${upmPackageURL}")
+        }
```

Comment:
This all seems to be designed for a single upm package reference.
I don't want to suggest to support multiple packages at the same time because that could lead to mayham.
But I would urge to read only the first line or the first line not starting with `#` or `//` etc.

```suggestion
        if (file.exists()) {
            def line = f.readLines().find {!it.startsWithAny("#", "//")}.trim()
            if(line.isEmpty() || !line.matches(/^https://.*?;\w+/) {
                //Log some warning
                //Line is either empty or doesn't match correct format
                //need logger
            } else {
                exists = true
                def parts = line.split(";")
                upmPackageURL = parts[0]
                upmPackageName  = parts[1]
                upmPackageNameUnversioned = upmPackageName.split("@")[0]
                upmPackageTar = new File ("${file.parentFile.path}/${upmPackageURL}")
            }
        }
```

One could of course just use a regex match to combine the check and the split.

Reply:
yes for sanity and pragmatic reasons it made sense to not need to complicated this even more with multiple packages. The ref file functionality is designed to be used only by WDKs/Libraries, not projects, and as such the 1:1 relationship between WDK <-> Wrapped UPM made sense.

Reply:
other comment: sure :)